### ISSUE TITLE
filename must be application when use git for repo.

### ISSUE THREAD

**Issue by dchack** [spring-cloud/spring-cloud-config#969]
I read many document and resource code, found it is not simple if i want change the filename about properties file on git,because  spring cloud config only load {application} whole words only matched filename.
the code is here:
org.springframework.cloud.config.server.environment.NativeEnvironmentRepository#getArgs

My question is why spring cloud config don't load all properties file by folder name or custom the filename? so, there are any way to solve this problem?

**Comment by spencergibb** [comment]
We're using standard spring boot mechanisms to load the properties. Not doesn't load by folder. What other name would you want other than application name?

**Comment by dchack** [comment]
@spencergibb tanks for your reply. When the properties become big,i just want to split the file to some small file.For example: application=message, my properties file like this: message-service. properties;
message-switch,message-globle. properties.but it's can't work.
And there is no way to change the mechanisms to load.I must put all properties in to one file.

**Comment by tdanylchuk** [comment]
you may use profiles for this purpose:

`service-name-switch.properties`
`service-name-globle.properties`

and in `service-name.properties` class add `spring.profiles.include=switch,globle`
in this case you will load all properties from general properties file + small ones

**Comment by dchack** [comment]
@tdanylchuk Yes, you are right.But there are tow points to change when add a new small properties.

**Comment by dchack** [comment]
@tdanylchuk @tdanylchuk hi,I found use `spring.profiles.include` can't work, when config client request server, the parameter "name" is use spring.cloud.name or spring.cloud.config.name.
So,I get an other way to fix it. spring.cloud.config.name=message,message-switch,and server will load two file: message. properties and message-switch. properties. But i don't know whether it's a best practice.

**Comment by tdanylchuk** [comment]
I've checked `spring.profiles.include` property and it works fine and as expected.
Can you show how it's not work for you?

Unfortunately I can't say what is bad\good practice, members could say more, but IMHO you should omit using `spring.cloud.config.name` in such particular case.

**Comment by dchack** [comment]
@spencergibb @tdanylchuk

**Comment by dchack** [comment]
@spencergibb @tdanylchuk 
hi, use` spring.cloud.config.name`is not very flexible,because it is in clients.
If i want add  properties name like : reds.properties;mysql.properties;memcache.properties.I can't change all clients properties.so i get a new way to fix it.
Add a `Interceptor` to config-server,and append string ",mysql,redis,memcache"to the parameter："name". so all "mysql,redis,memcache" properties will be load.
ok, it looks not good.Can we make `org.springframework.cloud.config.server.environment.NativeEnvironmentRepository#getArgs` to be public or give some way to custom the name about properties.

**Comment by tdanylchuk** [comment]
if you want to apply `reds.properties;mysql.properties;memcache.properties.` to all clients, you can create

`application-mysql.properties`
`application-memcache.properties`
and so on.

So if your service needs some of these properties - put profile in your `my-service.properties`
`spring.profiles.include=mysql,memcache`

Not all services needs to have e.g. `redis` properties, so if it need such - just add needed one to profile.

**Comment by shouwangqing** [comment]
I have the same problem with you, and I use @tdanylchuk 's method to solve it. but now how do you solve another type of configs, like logback-spring.xml，do you put the logback-spring.xml to the github or put the logback-spring.xml to the  resources path。do spring cloud config only manages *.properties file?
